### ISSUE TITLE
Support for a component version

### ISSUE THREAD

**Issue by julianoes** [mavlink/mavlink#1169]
It would be nice for a MAVLink component other than an autopilot to be able to announce its version. 

This is probably related to #1133.

**Comment by hamishwillee** [comment]
So a generic version of [AUTOPILOT_VERSION](https://mavlink.io/en/messages/common.html#AUTOPILOT_VERSION)?

**Comment by olliw42** [comment]
the general "problem" here is that components can be of various types, and that for "some components" this sort of message are already available, specifically for the camera there is already a CAMERA_INFORMATION message.

At some place there was a discussion if that should be extended to all components, or if there better should be messages for the various types of components. Given that CAMERA_INFORMATION is already existing and fixed part of the camera microservice, the answer probably would be to add a message for all non-camera and non-autopilot components.

I'd like to emphasize that as part of the proposal #1133, I had suggested a COMPONENT_INFORMATION message already, which might serve as an indication, see the related QGC issue https://github.com/mavlink/qgroundcontrol/issues/7503. 
The bottom line is that with progress with that proposal, also this issue here would be resolved.

**Comment by julianoes** [comment]
I think `COMPONENT_INFORMATION` messages would make sense. @olliw42 if you could make a pull request with the messages that would be great.

**Comment by olliw42** [comment]
I'm happy to try doing this

could you tell which ids I should use for the message and related request command (I used 700 in both...)

**Comment by auturgy** [comment]
Suggest close (https://github.com/mavlink/mavlink/pull/1176)

**Comment by hamishwillee** [comment]
@julianoes Is this alive, or something that will be part of component information?

**Comment by julianoes** [comment]
@hamishwillee good point. It should be part of the component information.

**Comment by hamishwillee** [comment]
Excellent. Now of course you get to make that actually happen :-)

**Comment by hamishwillee** [comment]
This got implemented as WIP, but is now being modified so that this information will be sent in the the component information metadata.
